tests/std/tests/P0896R4_ranges_alg_partial_sort_copy/test.cpp

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <span>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+using P = pair<int, int>;
+
+// Validate that partial_sort_copy_result aliases in_out_result
+STATIC_ASSERT(same_as<ranges::partial_sort_copy_result<int, double>, ranges::in_out_result<int, double>>);
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::partial_sort_copy(borrowed<false>{}, borrowed<false>{})),
+    ranges::partial_sort_copy_result<ranges::dangling, ranges::dangling>>);
+STATIC_ASSERT(same_as<decltype(ranges::partial_sort_copy(borrowed<false>{}, borrowed<true>{})),
+    ranges::partial_sort_copy_result<ranges::dangling, int*>>);
+STATIC_ASSERT(same_as<decltype(ranges::partial_sort_copy(borrowed<true>{}, borrowed<false>{})),
+    ranges::partial_sort_copy_result<int*, ranges::dangling>>);
+STATIC_ASSERT(same_as<decltype(ranges::partial_sort_copy(borrowed<true>{}, borrowed<true>{})),
+    ranges::partial_sort_copy_result<int*, int*>>);
+
+constexpr P source[]   = {{5, 11}, {1, 12}, {3, 13}, {4, 15}, {0, 16}, {2, 17}};
+constexpr P expected[] = {{0, 16}, {1, 12}, {2, 17}, {3, 13}, {4, 15}, {5, 11}};
+
+struct instantiator1 {
+    template <ranges::input_range In, ranges::random_access_range Out>
+    static constexpr void call() {
+#if !defined(__clang__) && !defined(__EDG__) // TRANSITION, VSO-938163
+#pragma warning(suppress : 4127) // conditional expression is constant
+        if ((!ranges::contiguous_range<In> && !ranges::contiguous_range<Out>) || !is_constant_evaluated())
+#endif // TRANSITION, VSO-938163
+        {
+            using ranges::partial_sort_copy, ranges::partial_sort_copy_result, ranges::equal, ranges::iterator_t,
+                ranges::less, ranges::min, ranges::next, ranges::size;
+
+            P output[2 * size(source)];
+            constexpr int sizes[] = {0, int{size(source) / 2}, int{size(source)}, int{2 * size(source)}};
+
+            { // Validate range overload
+                for (const int i : sizes) {
+                    In range1{source};
+                    Out range2{span{output}.first(static_cast<size_t>(i))};
+                    const same_as<partial_sort_copy_result<iterator_t<In>, iterator_t<Out>>> auto result =
+                        partial_sort_copy(range1, range2, less{}, get_first, get_first);

```

Comment:
Is there any consideration for testing different projections for the input and result ranges (e.g., `get_first` and `get_last`)?  I know that I had to carefully trace which projections were being called on what range elements as various heap helper functions were called, especially as sometimes they were passed in the order `_Proj2, _Proj1` that I had to verify was correct).

Reply:
Good point: I didn't have any such coverage because it's kid of a PITA to devise a case where heterogeneous projections make sense. I've contrived one ;)